manga109api/manga109api.py

```diff
@@ -17,22 +21,77 @@ def __init__(self, root_dir):
         with (self.root_dir / "books.txt").open("rt", encoding='utf-8') as f:
             self.books = [line.rstrip() for line in f]
 
-    def get_annotation(self, book, annotation_type="annotations"):
+
+    def get_annotation(self, book, annotation_type="annotations", separate_by_tag=True):
         """
-        Given a book title, return annotation in the form of dict.
+        Given a book title, return its annotations as a dict.
 
         Args:
-            book (str): A title of a book. Should be in self.books.
+            book (str): The title of the book to get the annotations of.
+                The title must be contained in the list `self.books`.
+            annotation_type (str) default `"annotations"` : The directory to load the xml data from.
+            separate_by_tag (bool) default `True` : When set to `True`, each annotation data type
+                ("frame", "face", "body", "text") will be stored in a different list in the output
+                dictionary. When set to `False`, all of the annotation data will be stored in a
+                single list in the output dictionary. In the latter case, the data in the list will
+                appear in the same order as in the original XML file.
 
         Returns:
-            annotation (dict): Annotation data consists of dict.
+            annotation (dict): The annotation data
         """
         assert book in self.books
-        with (self.root_dir / annotation_type / (book + ".xml")).open("rt", encoding= 'utf-8') as f:
-            annotation = xmltodict.parse(f.read())
-        annotation = json.loads(json.dumps(annotation))  # OrderedDict -> dict
-        annotation = _format_annotation(annotation)
-        _convert_str_to_int_recursively(annotation)  # str -> int, for some attributes
+
+        def int_literals_to_int(t):
+            # Convert certain keys to integer types.
+            # The following keys will be affected with this function:
+            # '@index', '@width', '@height', '@xmax', '@ymax', '@xmin', '@ymin'
```

Comment:
As @ku21fan suggests, can you write some comments and examples?

Reply:
Thanks, added a more detailed description of the function.